### DISCUSSION TITLE
Finding rotation and translation of camera from projection matrix

### DISCUSSION THREAD

**Discussion by Moj-Dev** [microsoft/AirSim#3302] [Support Q&A]
I used client.simGetCameraInfo(str(camera_name)).proj_mat to find the projection matrix. How can I calculate R and T from the projection matrix?  Can I use  "cv2.decomposeProjectionMatrix" ?
 Also, I obtained R and T like this:
 -------------------------------------------------------------------------------------------------------------------------------------------------------
         camera_info = self.client.simGetCameraInfo(str(0)).pose.orientation
        q0 = camera_info.w_val
        q1 = camera_info.x_val
        q2 = camera_info.y_val
        q3 = camera_info.z_val
        r00 = 2 * (q0 * q0 + q1 * q1) - 1
        r01 = 2 * (q1 * q2 - q0 * q3)
        r02 = 2 * (q1 * q3 + q0 * q2)
        r10 = 2 * (q1 * q2 + q0 * q3)
        r11 = 2 * (q0 * q0 + q2 * q2) - 1
        r12 = 2 * (q2 * q3 - q0 * q1)
        r20 = 2 * (q1 * q3 - q0 * q2)
        r21 = 2 * (q2 * q3 + q0 * q1)
        r22 = 2 * (q0 * q0 + q3 * q3) - 1
        R = np.array([[r00, r01, r02],[r10, r11, r12],[r20, r21, r22]])
        T = self.client.simGetCameraInfo(str(0)).pose.position
 -------------------------------------------------------------------------------------------------------------------------------------------------------
Is it correct?

**Comment by ahmed-elsaharti** [comment]
@Moj-Dev Based on [Wikipedia](https://en.wikipedia.org/wiki/Quaternions_and_spatial_rotation#Quaternion-derived_rotation_matrix) and [this article](https://www.euclideanspace.com/maths/geometry/rotations/conversions/quaternionToMatrix/index.htm) it looks like r00, r11 and r22 are off.
Also, another way of grabbing the rotation matrix corresponding to a quaternion would be to use the [Scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html) library
